app/src/main/java/com/rootstrap/android/util/permissions/PermissionFragment.kt

```diff
@@ -0,0 +1,70 @@
+package com.rootstrap.android.util.permissions
+
+import android.content.pm.PackageManager
+import com.rootstrap.android.ui.base.BaseFragment
+
+open class PermissionFragment : BaseFragment() {
+
+    private var permissionListener: PermissionResponse? = null
+
+    private fun requestPermission(permissions: Array<String>) {
+        activity?.let { activityContext ->
+            val notGrantedPermissions = activityContext.checkNoGrantedPermissions(permissions)
+
+            when {
+                notGrantedPermissions.isEmpty() -> permissionListener?.granted()
+                else -> requestPermissions(
+                    notGrantedPermissions.toTypedArray(),
+                    REQUEST_PERMISSION_REQUEST_CODE
+                )
+            }
+        }
+    }
+
+    fun checkPermission(permission: String, listener: PermissionResponse) {
+        permissionListener = listener
+        requestPermission(arrayOf(permission))
+    }
+
+    fun checkPermissions(permissions: Array<String>, listener: PermissionResponse) {
+        permissionListener = listener
+        requestPermission(permissions)
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        permissionListener?.let { listener ->
+            if (requestCode != REQUEST_PERMISSION_REQUEST_CODE) {
+                return
+            }
+
+            val permissionDenied = mutableListOf<String>()
+
+            for (i in grantResults.indices) {
+                if (grantResults[i] != PackageManager.PERMISSION_GRANTED) {
+                    permissionDenied.add(permissions[i])
+                }
+            }
+
+            val granted = permissionDenied.size == 0
+
+            when {
+                granted -> listener.granted()
+                else -> {
+                    for (s in permissionDenied) {
```

Comment:
Same here, we should give the constant a more descriptive name than `s`. It should be called `deniedPermission`.